src/MiniProfiler.Shared/ProfileProviders/BaseProfilerProvider.cs

```diff
@@ -29,6 +29,11 @@ public abstract class BaseProfilerProvider : IProfilerProvider
         public abstract MiniProfiler GetCurrentProfiler();
 
         /// <summary>
+        /// GEts or sets the current head timing. This is used by <see cref="MiniProfiler.Head"/>.
+        /// </summary>
+        public abstract Timing CurrentHead { get; set; }
```

Comment:
The timing is specific to the operation, so yes, possibly making the head AsyncLocal makes the most sense. We'll have to think through if that actually breaks anything though. And the context fetch isn't free either for all requests, so it's not a clear win on the performance side.